### PR TITLE
feat: add API and software version HTTP headers

### PR DESCRIPTION
**Pull request by rhyslbw**
# Context
We're currently missing metadata in the HTTP requests to assist with managing multiple API versions at any given time. There's also no indication of the version of the SDK package being used to make the request.

# Proposed Solution
Each request made by the `HttpProvider` client now includes the API version, sourced from the top-level `cardano-services` `ProviderServer`, as a declaration of compatibility, and the software version for context. This can be used to dynamically route traffic to compatible servers, with forwards compatibility considered.

# Important
This change needs to be coordinated with backend deployments after merge.

### CODE REVIEW

File: packages/cardano-services/test/Utxo/UtxoHttpService.test.ts [hunk 1]
```diff
@@ -12,13 +12,15 @@ import { OgmiosCardanoNode } from '@cardano-sdk/ogmios';
...
 import { healthCheckResponseMock, mockCardanoNode } from '../../../core/test/CardanoNode/mocks';
+import { loadApiVersion } from '../util';
 import { logger } from '@cardano-sdk/util-dev';
 import axios from 'axios';
 
 const APPLICATION_JSON = 'application/json';
 const APPLICATION_CBOR = 'application/cbor';
 const UNSUPPORTED_MEDIA_STRING = 'Request failed with status code 415';
 const BAD_REQUEST_STRING = 'Request failed with status code 400';
+const version = { api: loadApiVersion(), software: '0.0.0' };
```
Comment (mkazlauskas) [inline packages/cardano-services/test/Utxo/UtxoHttpService.test.ts:23]: This could be reused from a test `util.ts` as it's the same for all providers

**Review by mkazlauskas — CHANGES_REQUESTED** [review]

File: packages/cardano-services-client/src/util.ts [hunk 2]
```diff
@@ -0,0 +1,10 @@
+import fs from 'fs';
+import path from 'path';
+
+const loadJson = (relativePath: string[]) =>
+  JSON.parse(fs.readFileSync(path.join(__dirname, ...relativePath), 'utf-8'));
```
[the remark this answers is not in the document]
Comment (rhyslbw) [reply packages/cardano-services-client/src/util.ts]: Nope! Good catch https://github.com/input-output-hk/cardano-js-sdk/pull/634/commits/4f278a7cad4608157d0381a5c1ed719d7f3f7a33

File: packages/cardano-services/test/Utxo/UtxoHttpService.test.ts (hunk 1 above)
Comment (rhyslbw) [reply packages/cardano-services/test/Utxo/UtxoHttpService.test.ts]: https://github.com/input-output-hk/cardano-js-sdk/pull/634/commits/36fa5abb7db75b5a8fd8eaeded9997e731fa20e0

**Review by iccicci — APPROVED** [review]
GJ

**Review by mirceahasegan — APPROVED** [review]
Great work, @rhyslbw ! 🚀

**Review by mkazlauskas — DISMISSED** [review]
Nice work! :rocket:

File: packages/cardano-services-client/src/version.ts [hunk 3]
```diff
@@ -2,5 +2,5 @@
 // auto-generated using /home/rhys/code/iohk/cardano-js-sdk/packages/cardano-services-client/scripts/createVersionSource.js
 export const version = {
   api: '1.0.0',
-  software: '0.7.2'
+  software: '0.9.8'
```
Comment (mirceahasegan) [inline packages/cardano-services-client/src/version.ts:5]: Is this file suppose to be part of the submitted files? I saw it was gitignored :thinking:

**Review by mirceahasegan — APPROVED** [review]
Fantastic work, @rhyslbw ! :rocket:

**Review by gytis-ivaskevicius — COMMENTED** [review]
Didn't we decide to go with a string base path parameter? I'm afraid that edge cases will come up which will not be covered by just having version hardcoded

File: packages/cardano-services-client/src/version.ts (hunk 3 above)
Comment (rhyslbw) [reply packages/cardano-services-client/src/version.ts:5]: Good catch, I'll purge it.

**Review by iccicci — DISMISSED** [review]
Great!

File: packages/cardano-services-client/src/version.ts (hunk 3 above)
Comment (rhyslbw) [reply packages/cardano-services-client/src/version.ts:4]: Decided to keep it in source control, so removed from the ignore file.

**Review by mirceahasegan — DISMISSED** [review]

**Review by mirceahasegan — APPROVED** [review]

**Pull request merged**